packages/network-controller/src/NetworkController.ts

```diff
@@ -212,8 +212,11 @@ export type NetworkConfiguration = {
  * Custom RPC endpoints do not need a `networkClientId` property because it is
  * assumed that they have not already been added and therefore network clients
  * do not exist for them yet (and hence IDs need to be generated).
+ *
+ * However Custom RPC endpoints, that are synchronized between devices,
+ * can contain a `networkClientId` set on both devices.
```

Comment:
For custom RPCs, the `networkClientId` is a random UUID.
To make it easier for syncing, if a device is adding new networks from remote, we can reuse the UUID.

This was there is less chance of having some weird state where 2 devices use the same networks + RPCs, but they are different UUIDs.

Reply:
This is mostly optional (to keep uuid consistency) across devices.